extensions/amp-minute-media-player/0.1/amp-minute-media-player.js

```diff
@@ -0,0 +1,472 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {Layout} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {VideoEvents} from '../../../src/video-interface';
+import {addParamsToUrl} from '../../../src/url';
+import {
+  createFrameFor,
+  isJsonOrObj,
+  mutedOrUnmutedEvent,
+  objOrParseJson,
+  originMatches,
+  redispatch,
+} from '../../../src/iframe-video';
+import {dev, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {
+  fullscreenEnter,
+  fullscreenExit,
+  isFullscreenElement,
+  removeElement,
+} from '../../../src/dom';
+import {getData, listen} from '../../../src/event-helper';
+import {installVideoManagerForDoc} from '../../../src/service/video-manager-impl';
+
+/** @const */
+const TAG = 'amp-minute-media-player';
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpMinuteMediaPlayer extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    // Declare instance variables with type annotations.
+    /** @private {Element} */
+    this.iframe_ = null;
+
+    /** @private {string} */
+    this.contentType_ = '';
+
+    /** @private {?string} */
+    this.playerId_ = null;
+
+    /** @private {?string} */
+    this.contentId_ = '';
+
+    /** @private {?string} */
+    this.scannedElementType_ = '';
+
+    /** @private {?string} */
+    this.scannedElement_ = '';
+
+    /** @private {?string} */
+    this.tags_ = '';
+
+    /** @private {?string} */
+    this.minimumDateFactor_ = '';
+
+    /** @private {?string} */
+    this.scopedKeywords_ = '';
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {?boolean}  */
+    this.muted_ = false;
+  }
+
+  /**
+   * @param {boolean=} onLayout
+   * @override
+   */
+  preconnectCallback(onLayout) {
+    this.preconnect.url(this.iframeSource_());
+    // Host that serves player configuration and content redirects
+    this.preconnect.url('https://www.oo-syringe.com', onLayout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.initFields_();
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return (
```

Comment:
Any particular reason why you're not using `isLayoutSizeDefined()` from `src/layout.js`?

Reply:
Yes, according to the [documentation](https://github.com/ampproject/amphtml/blob/master/contributing/building-an-amp-extension.md#what-layout-should-your-element-support), this function doesn't support two layouts: **flex-item & nodispaly** and we would like to support those layouts in our implementation.

Reply:
1. Why `nodisplay`?
2. Combine `flex-item` with `isLayoutSizeDefined()` instead of concatenating many via `||`. Using `isLayoutSizeDefined` is for forward-compatibility.

```js
return (
  isLayoutSizeDefined(layout) ||
  layout == LAYOUT.FLEX_ITEM
);
```